modules/layers/src/text-layer/text-layer.js

```diff
@@ -320,6 +336,10 @@ export default class TextLayer extends CompositeLayer {
     const CharactersLayerClass = this.getSubLayerClass('characters', MultiIconLayer);
     const BackgroundLayerClass = this.getSubLayerClass('background', TextBackgroundLayer);
 
+    if (data.attributes && data.attributes.instancePickingColors) {
+      this._expandPickingColors();
```

Comment:
TextLayer is behaving as expected. If you manipulate the attribute it needs to be done outside of the layer.

Reply:
For the manipulation I need the `startIndices` that are calculated by `TextLayer`. Are you saying that I should duplicate this calculation inside `GeojsonLayer`?

Reply:
What you are doing here will break the apps that are currently providing `instancePickingColors` correctly.

Reply:
I've moved the conversion to GeoJsonLayer